### PR TITLE
Fix flaky `ProviderDetailsCreator` spec

### PR DESCRIPTION
**Pull request by cpjmcquillan**
This test file contained a [flaky spec].

Trying to isolate the flaky spec in order to reproduce and fix the issue proved difficult, so I took a more pragmatic approach to resolve the issue.

1. Ensure each test is self-contained
2. Ensure each test is repeatable (i.e deterministic)
3. Avoid using `let!` (linked to step 1.)

In the process of refactoring the tests, it also highlighted some tests that didn't add any value -those have been removed.

I'm confident the refactored tests are not flaky, and will improve the robustness of our test suite -saving us frustration, time, and money.

[flaky spec]: https://app.circleci.com/pipelines/github/ministryofjustice/laa-apply-for-legal-aid/18165/workflows/ae431d1e-b060-4638-8b78-91df79353cc1/jobs/95186/steps

### CODE REVIEW

File: spec/services/provider_details_creator_spec.rb [hunk 1]
```diff
@@ -4,177 +4,297 @@
...
-      it "updates the contact_id of the provider" do
-        expect { subject }.to change { third_provider.reload.contact_id }.to(api_response[:contacts][2][:id])
+        expect(provider).to have_attributes(
+          name: "",
+          contact_id: 104,
+        )
       end
     end
 
-    context "when the emails match" do
```
Comment (cpjmcquillan) [inline spec/services/provider_details_creator_spec.rb:97]: This test did not test what it said it did.

Provider details are matched by `username` comparison, not email.

File: spec/services/provider_details_creator_spec.rb [hunk 2]
```diff
@@ -4,177 +4,297 @@
...
+          have_attributes(
+            class: Office,
+            ccms_id: ccms_office2.id.to_s,
+            code: ccms_office2.code,
+          ),
+        )
       end
-    end
 
-    context "when the username matches" do
```
Comment (cpjmcquillan) [inline spec/services/provider_details_creator_spec.rb:87]: This test didn't test anything different to the test on line 82.

File: spec/services/provider_details_creator_spec.rb [hunk 3]
```diff
@@ -4,177 +4,297 @@
...
+          provider:,
+          firm: ccms_firm,
+          offices: [ccms_office1, ccms_office2],
+        )
+
+        described_class.call(provider)
 
-    context "when the names match" do
-      it "updates the contact_id of the provider" do
-        expect { subject }.to change { provider.reload.contact_id }.to(api_response[:contacts][0][:id])
```
Comment (cpjmcquillan) [inline spec/services/provider_details_creator_spec.rb:83]: And this test was the same as the test on line 77.

**Comment by cpjmcquillan** [comment]
Here's [another flaky test run](https://app.circleci.com/pipelines/github/ministryofjustice/laa-apply-for-legal-aid/18258/workflows/c972a216-0a76-4c00-9592-91ba734d6ef9/jobs/95635) that made me grumpy and sad and wasted build time 😢

**Review by RoseSAK — APPROVED** [review]
I'm not sure where we ended up in the whole discussion of how to write rspec tests and keep the test suite consistent (I think we never managed to have a whole team discussion as various people weren't there each time?) but I am happy to approve this if it fixes the flaky tests 🎉 all looks good to me 👍 always good to get more opinions though

**Comment by cpjmcquillan** [comment]
As DJ Khaled says...

**[another one](https://app.circleci.com/pipelines/github/ministryofjustice/laa-apply-for-legal-aid/18306/workflows/9ec2c327-25b6-4f01-a3e1-51a3e3b92d5c/jobs/95865)**

😢

**Pull request merged**